### PR TITLE
Improve citation validation with parameter tuning

### PR DESCRIPTION
**Pull request by ChenKua**
# Your checklist for this pull request
Thank you for submitting a pull request! To speed up the review process, please follow this checklist:

- [x] My Pull Request is small and focused on one topic so it can be reviewed easily 
- [x] My code follows the style guidelines of this project (`make format`)
- [x] Commit messages are detailed
- [x] I have performed a self-review of my code
- [x] I commented hard-to-understand parts of my code
- [x] I updated the documentation (docstrings, `/docs`)
- [x] My changes generate no new warnings (or explain any new warnings and why they're ok)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass when I run `pytest tests` (offline mode)

Additional steps for code with networking dependencies:

- [ ] I followed the [offline and online testing guidelines](https://sherpa-ai.readthedocs.io/en/latest/How_To/Test/test.html#offline-and-online-testing)
- [ ] All tests pass when I run `pytest tests --external_api` (online mode, making network calls)

# Description
Describe your pull request here.

What does this PR implement or fix? Explain.

If this PR resolves any currently open issues then mention them like this: `Closes #xxx`.
Github will close such issues automatically when your PR is merged into `main`.

Any relevant logs, error output, etc?

Any other comments? For example, will other contributors need to install new libraries via `poetry install` after picking up these changes?

💔Thank you!

### CODE REVIEW

**Comment by ChenKua** [comment]
@20001LastOrder can you review the samll changes to citation validation?

File: src/tests/unit_tests/test_citation_validation_fever.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    text = "Nikolaj Coster-Waldau was in a suspense movie of Danish origin."
+    module = CitationValidation()
+    result = module.parse_output(text, resource)
+    print(result)
+
+    # 8
+    text = "Nikolaj Coster-Waldau is an actor."
+    module = CitationValidation()
+    result = module.parse_output(text, resource)
+    print(result)
```
Comment (20001LastOrder) [inline src/tests/unit_tests/test_citation_validation_fever.py:73]: You can remove these prints, or use loguru instead

File: src/tests/unit_tests/test_citation_validation_fever.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+from sherpa_ai.output_parsers.citation_validation import CitationValidation
+
+
+def test_citation_validation():
```
Comment (20001LastOrder) [inline src/tests/unit_tests/test_citation_validation_fever.py:4]: Please put these test cases also in the `test_citation_validation` file. Maybe have a comment saying these examples are from the fever dataset (You can also add a link to the dataset). Currently, it looks a bit confusing if a person is not familiar with the fever dataset.

**Review by 20001LastOrder — CHANGES_REQUESTED** [review]
Thanks for the PR, please see these few suggestions.

File: src/tests/unit_tests/test_citation_validation_fever.py (hunk 2 above)
Comment (ChenKua) [reply src/tests/unit_tests/test_citation_validation_fever.py:4]: remove. This file is for some random testing.

**Review by 20001LastOrder — APPROVED** [review]

**Pull request merged**